crates/bevy_ecs/src/component.rs

```diff
@@ -138,6 +138,24 @@ impl ComponentInfo {
     }
 }
 
+/// A [`ComponentId`] is an semi-opaque value which uniquely identifies the type of
+/// a [`Component`] within a [`World`](crate::world::World).
```

Comment:
```suggestion
/// Semi-opaque value which uniquely identifies the type of
/// a [`Component`] within a [`World`](crate::world::World).
```

It's not needed to repeat the name of the involved item. It just makes the description longer.

I don't have strong opinions about the “semi-opaque” terminology, but it seems to me some kind of undefined, relative language that convey no information.

Reply:
It was originally just opaque, but I threw the "semi-" in there when I realized the `ComponentId::new` and `ComponentId::index`  functions are public. I'd still like to convey that a typical user should treat a `ComponentId` as opaque, but I agree that the current phrasing isn't ideal.